cmd/containerboot/main.go

```diff
@@ -253,7 +251,7 @@ func main() {
 		return nil
 	}
 
-	if !cfg.AuthOnce {
+	if !runTailscaleSet(cfg) {
```

Comment:
It's a bit weird that running `tailscale up` is gated by a function called `run tailscale set`. Perhaps it would be better to leave this as `!cfg.AuthOnce`, since `authTailscale` exits early if we use a config file anyway?

Reply:
This PR adds a bunch more combinations of 'is/is not `cfg.AuthOnce` and <something else>' to the extent that I think it becomes difficult to read through. I am trying to distinguish between the three install modes (run tailscaled + tailscale auth, run tailscaled + tailscale auth and subsequent tailscale set, run tailscaled only).
I've updated the helper functions to have hopefully better names, happy to rename them differently if you have other suggestions https://github.com/tailscale/tailscale/pull/10759/commits/0bf017c87cb4c704fa812ca28dc78b1fa5bb8cb0#diff-700830ac79a41da42cd3a82065f0fee3003a4390eead4db1de4bf30557a06bd6L999-R1019